### PR TITLE
Add analytics to connect-kit

### PR DESCRIPTION
**Pull request by RamyEB**
## Main Goal: 

Be able to retrieve usage metrics for the connect kit

## Why: 

In order to effectively prioritize tasks in regards to the connect kit, we need to be able to retrieve usage data for the connect kit. For example one of the main difficulties we have is that we do not know which Dapps support the connect kit. 

## What data:

There are different types of information we would like to be able to obtain such as:

Provenance → Which Dapp is a user accessing the connect kit from? 

Flow → What are the steps the user follows, does he exit the modal without finalizing the connection? 

Usage → How many users go through wallet connect versus the extension?  (daily, weekly monthly)

Session length & frequency 

---

- [JIRA ticket for this PR.]( https://ledgerhq.atlassian.net/jira/software/c/projects/LIVE/boards/623?modal=detail&selectedIssue=LIVE-7781)

Tasks : 
- [x] Initialize Segment 
- [x] Connect Segment account and Mixpanel (or another panel)
- [x] Provenance metric 
- [x] Flow
- [ ] Usage
- [ ] Session length 

Stack : 
Segment 

ressources : 
JIRA ticket : https://ledgerhq.atlassian.net/jira/software/c/projects/LIVE/boards/623?modal=detail&selectedIssue=LIVE-7781

### CODE REVIEW

**Comment by hlopes-ledger** [comment]
@RamyEB, did you forget to push some commit?

You mentioned moving the analytics code from the connect-kit-loader package to the connect-kit one but I still see it on the loader. I think it would make more sense for it to be in CK as that's where we will want to track the events. On the CK README we recommend people to use the loader but there might be cases in which the use CK directly, and in those cases there will be no analytics if they don't use the loader.

We had also talked about seting up the Segment key as a secret using rollup-plugin-dotenv and updating the .env file on the GitHub deployment action but I don't see that in the PR.

I also think the PR should have the steps needed to test it. Should we create a segment account to test the PR locally? What are all the steps to confirm it is working? That description should then be added to the Confluence page or the README file.

**Comment by RamyEB** [comment]
> @RamyEB, did you forget to push some commit?
> 
> Seems like I looked only at part of the changes, scratched my comments below.
> 
> ~You mentioned moving the analytics code from the connect-kit-loader package to the connect-kit one but I still see it on the loader. I think it would make more sense for it to be in CK as that's where we will want to track the events. On the CK README we recommend people to use the loader but there might be cases in which the use CK directly, and in those cases there will be no analytics if they don't use the loader.~
> 
> ~We had also talked about seting up the Segment key as a secret using rollup-plugin-dotenv and updating the .env file on the GitHub deployment action but I don't see that in the PR.~
> 
> I also think the PR should have the steps needed to test it. Should we create a segment account to test the PR locally? What are all the steps to confirm it is working? That description should then be added to the Confluence page or the README file.

1. Create a segment account 
2. Create a new source
3. Copy your key into a `.env` file at root

File: packages/connect-kit/src/components/UseLedgerLiveModal/UseLedgerLiveModal.tsx [hunk 1]
```diff
@@ -59,10 +63,26 @@ const UseLedgerLiveModal = ({
...
     window.open('https://www.ledger.com/ledger-live');
     return false;
   }, []);
 
+
+
+  useEffect(() => {
+    init(
+      {
+        dappName: "connect-kit",
```
Comment (hlopes-ledger) [inline packages/connect-kit/src/components/UseLedgerLiveModal/UseLedgerLiveModal.tsx:76]: cc @ComradeAERGO

do you know if we will use the same instance (SEGMEN_KEY) that Ledger uses for other Ledger projects?
In that case, is this intended to filter the Connect Kit entries from the other apps? Is dappName appropriate or is there already a prop used to distinguish the other apps?

File: .github/workflows/release_connect-kit.yml [hunk 2]
```diff
@@ -31,8 +31,11 @@ jobs:
 
       # Build
       - name: Build
+        env:
+          MY_SECRET: ${{ secrets.SEGMENT_KEY }}
```
Comment (hlopes-ledger) [inline .github/workflows/release_connect-kit.yml:35]: Do we know the secret key already?
Do we add it ourselves to the repository secrets or is this done by the infra team?

**Comment by hlopes-ledger** [comment]
LGTM

I think it will need some adjustments to track when Connect Kit returns a provider, which happens outside of the react scope and we won't be able to use the hooks, but I suppose that can be done later.

**Review by hlopes-ledger — APPROVED** [review]

File: packages/connect-kit/package.json [hunk 3]
```diff
@@ -51,5 +51,9 @@
     "tslib": "^2.4.0",
     "typescript": "^4.7.3"
   },
-  "dependencies": {}
+  "dependencies": {
+    "@segment/analytics-next": "^1.53.0",
+    "rollup-plugin-dotenv": "^0.5.0",
+    "uuid": "^9.0.0"
```
Comment (hlopes-ledger) [inline packages/connect-kit/package.json:57]: I've had to also add the `@types/uuid` package for CK to compile

File: examples/direct-integration/src/App.js [hunk 4]
```diff
@@ -25,11 +25,10 @@ export default function Home() {
 
   const connectWallet = async () => {
     console.log('> connectWallet');
-
     resetState();
 
     try {
-      const connectKit = await loadConnectKit();
+      const connectKit = await loadConnectKit();
```
Comment (hlopes-ledger) [inline examples/direct-integration/src/App.js:31]: does not seem like an intended change

File: examples/direct-integration/src/App.js [hunk 5]
```diff
@@ -55,7 +54,6 @@ export default function Home() {
       }
     } catch (error) {
       console.error(error)
-      setMessage(error.message);
```
Comment (hlopes-ledger) [inline examples/direct-integration/src/App.js:58]: do you want to hide the error from the user?

**Pull request merged**